service/builder/exporters_builder.go

```diff
@@ -332,3 +357,18 @@ func createMetricsExporter(factoryBase component.ExporterFactoryBase,
 	// use ExporterFactoryOld.CreateMetricsExporter.
 	return factoryBase.(component.ExporterFactoryOld).CreateMetricsExporter(logger, cfg)
 }
+
+// createLogExporter creates a data exporter based on provided factory type.
+func createLogExporter(
+	factoryBase component.ExporterFactoryBase,
+	logger *zap.Logger,
+	cfg configmodels.Exporter,
+) (component.LogExporter, error) {
+	if factory, ok := factoryBase.(component.LogExporterFactory); ok {
```

Comment:
I usually prefer to check for error conditions not for non-errors.

Reply:
Done.